editoast/src/views/infra/mod.rs

```diff
@@ -350,21 +350,30 @@ async fn get_speed_limit_tags(
     ))
 }
 
-/// Returns the set of voltages for a given infra
+#[derive(Debug, Clone, Deserialize)]
+struct GetVoltagesQueryParams {
+    #[serde(default)]
+    include_rolling_stock_modes: bool,
+}
+
+/// Returns the set of voltages for a given infra and/or rolling_stocks modes.
+/// If include_rolling_stocks_modes is true, it returns also rolling_stocks modes.
 #[get("/voltages")]
-async fn get_voltages(infra: Path<i64>, db_pool: Data<DbPool>) -> Result<Json<Vec<f64>>> {
+async fn get_voltages(
+    infra: Path<i64>,
+    param: Query<GetVoltagesQueryParams>,
+    db_pool: Data<DbPool>,
+) -> Result<Json<Vec<String>>> {
     let infra = infra.into_inner();
+    let include_rolling_stock_modes = param.into_inner().include_rolling_stock_modes;
     let voltages: Vec<Voltage> = block::<_, Result<_>>(move || {
         let mut conn = db_pool.get()?;
-        match sql_query(
-            "SELECT DISTINCT ((data->'voltage')->>0)::float AS voltage
-                FROM osrd_infra_catenarymodel
-                WHERE infra_id = $1
-                ORDER BY voltage",
-        )
-        .bind::<BigInt, _>(infra)
-        .load(&mut conn)
-        {
+        let query = if !include_rolling_stock_modes {
```

Comment:
```suggestion
        let query = if !include_rolling_stock_modes {
```
Wow I didn't know we could do smth like this, thanks !